lib/notifications/client/speaker.rb

```diff
@@ -49,15 +49,11 @@ def post(kind, form_data)
       # @param id [String]
       # @param options [Hash] query
       # @see #perform_request!
-      def get(id = nil, options = nil)
-        if !options.nil? && options.keys.any?
-          path = "?" + URI.encode_www_form(options)
-        end
-
-        request = Net::HTTP::Get.new(
-          "#{BASE_PATH}/#{id}#{path}",
-          headers
-        )
+      def get(id = nil, options = {})
+        path = BASE_PATH.dup
+        path << "/" << id if id
+        path << "?" << URI.encode_www_form(options) if options.any?
+        request = Net::HTTP::Get.new(path, headers)
         perform_request!(request)
       end
```

Comment:
I'm not happy about hand rolling this, however, prefer that over adding another gem to do it for me - and as such a dependency to every project it is used in. 

My ultimate thought is that this class should be re-written and simplified - as it feels this could be done far easier - but the current design works for it's intended purpose. 

Given this code is unlikely to ever change - now that it is working - I would argue it's alright to leave as is for now, and rewrite if any changes are needed in the future.